solang-parser/src/solidity.lalrpop

```diff
@@ -752,8 +752,10 @@ NonIfStatement: Statement = {
     <l:@L> "unchecked" "{" <statements:Statement*> "}" <r:@R> => {
         Statement::Block { loc: Loc::File(file_no, l, r), unchecked: true, statements }
     },
-    <l:@L> "assembly" <assembly:AssemblyBlock> <r:@R> => {
-        Statement::Assembly { loc: Loc::File(file_no, l, r), assembly }
+    <l:@L> "assembly" <dialect:string?> <assembly:AssemblyBlock> <r:@R> => {
+        Statement::Assembly {
+         loc: Loc::File(file_no, l, r), assembly,
+          dialect: if let Some(dialect_name) = dialect {Some(dialect_name.to_owned())} else {None}}
```

Comment:
You can write the if statement as `dialect.map(|dialect| dialect.to_owned())`, much shorter.

The indentation is askew.